packages/debug/configMapRegistryWatcher.go

```diff
@@ -0,0 +1,80 @@
+package debug
+
+import (
+	"context"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"time"
+)
+
+var tickerInterval = time.Second * 30
+
+type defaultConfigMapRegistryWatcher struct {
+	configMapInterface         configMapInterface
+	configMapDebugModeRegistry debugModeRegistry
+}
+
+// NewDefaultConfigMapRegistryWatcher creates an instance of defaultConfigMapRegistryWatcher.
+func NewDefaultConfigMapRegistryWatcher(configMapInterface configMapInterface, registry debugModeRegistry) *defaultConfigMapRegistryWatcher {
+	return &defaultConfigMapRegistryWatcher{
+		configMapInterface:         configMapInterface,
+		configMapDebugModeRegistry: registry,
+	}
+}
+
+// StartWatch checks if the disableAtTimestamp in the registry is after now and if yes it disables the debug mode.
+func (w *defaultConfigMapRegistryWatcher) StartWatch(ctx context.Context) {
+	go func() {
+		w.doWatch(ctx)
+	}()
+}
+
+func (w *defaultConfigMapRegistryWatcher) doWatch(ctx context.Context) {
+	ticker := time.NewTicker(tickerInterval)
+	for {
```

Comment:
🚫 **[checkstyle]** <gosimple> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>S1000: should use for range instead of for { select {} }